crates/knope-versioning/src/versioned_file/mod.rs

```diff
@@ -393,15 +416,37 @@ pub struct Config {
     pub(crate) format: Format,
     /// If, within the file, we're versioning a dependency (not the entire package)
     pub dependency: Option<String>,
+    /// If set, use regex pattern matching to find and replace the version
+    pub regex: Option<String>,
 }
 
 impl Config {
     /// Create a verified `Config` from a `RelativePathBuf`.
     ///
     /// # Errors
     ///
-    /// If the file name does not match a supported format
-    pub fn new(path: RelativePathBuf, dependency: Option<String>) -> Result<Self, UnknownFile> {
+    /// If the file name does not match a supported format and no regex is provided
+    /// If both dependency and regex are provided
+    pub fn new(
+        path: RelativePathBuf,
+        dependency: Option<String>,
+        regex: Option<String>,
+    ) -> Result<Self, UnknownFile> {
+        // Check that both dependency and regex are not set
+        if dependency.is_some() && regex.is_some() {
+            return Err(UnknownFile { path });
```

Comment:
I don't think this error is going to be clear enough to users. We should be explicit about what the problem is.